torch/csrc/autograd/VariableTypeManual.cpp

```diff
@@ -139,7 +140,7 @@ Tensor & copy_(Tensor & self, const Tensor & src, bool non_blocking) {
   check_inplace(self);
   std::shared_ptr<CopyBackwards> grad_fn;
   auto requires_grad = compute_requires_grad(self, src);
-  requires_grad &= isFloatingPoint(self.scalar_type());
+  requires_grad &= isFloatingType(self.scalar_type()) || isComplexType(self.scalar_type());
```

Comment:
hmmm why should complex int be requiring grad?

Reply:
So, when I looked, isComplexType tests positive only for ComplexHalf, ComplexFloat and ComplexDouble. That sounded continuous to me, but I could be wrong.

Reply:
I agree that those should be requiring grad. However, `isComplexType` seems too general and could in future include `ComplexLong` for example. Are these complex types are not considered as floating types in `isFloatingType`?

Reply:
I based the code on what is in ScalarType.h:

https://github.com/pytorch/pytorch/blob/c813503f057db361eb44740e7a6932ddf3fb450d/c10/core/ScalarType.h#L267-L277

But I don't have a strong opinion either way, so I'd be happy to take it out if you prefer.

Reply:
I don't either. But I think it might make most sense to leave the original behavior (ignoring complex types) unchanged. A comment explaining the complex situation would be extra nice. :)

Reply:
Are there complex integral types? 😮

Reply:
I guess you could call Gaussian integers that... I'm not sure if we're really going to add support for that, but I agree that adding a comment that "`isComplexType(t)` implies that `t` is dense" next to `isComplexType` might be a good idea.

Reply:
Well yeah, fixing isComplexType is great, but let's make this PR about bfloat16.